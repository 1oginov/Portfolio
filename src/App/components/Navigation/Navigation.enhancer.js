/* @flow */

import { connect } from 'react-redux';

const mapStateToProps = ({ app: { route } }) => ({ route });

export default connect(mapStateToProps);
