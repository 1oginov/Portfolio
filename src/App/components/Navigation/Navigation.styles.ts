import { withStyles } from '@material-ui/core';

export default withStyles({
  bottomMock: {
    height: 56,
  },
  bottomNavigation: {
    bottom: 0,
    left: 0,
    position: 'fixed',
    right: 0,
  },
});
