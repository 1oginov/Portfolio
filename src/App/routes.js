/* @flow */

export const CREATE_POSITION = 'CREATE_POSITION';
export const HOME = 'HOME';
export const PROFILE = 'PROFILE';
