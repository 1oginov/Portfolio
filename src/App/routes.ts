export const CREATE_POSITION = 'CREATE_POSITION';
export const HOME = 'HOME';
export const POSITION = 'POSITION';
export const PROFILE = 'PROFILE';
