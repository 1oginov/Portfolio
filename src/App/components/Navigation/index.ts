import Navigation from './Navigation';
import enhance from './Navigation.enhancer';

export default enhance(Navigation);
