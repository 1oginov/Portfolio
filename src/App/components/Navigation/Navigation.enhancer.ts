import * as React from 'react';
import { connect } from 'react-redux';
import { compose, lifecycle, withProps } from 'recompose';

import State from 'State';

import { navigate, NavigateAction } from '../../actions';
import { Props } from './Navigation';
import * as R from '../../routes';

interface StateProps {
  route: R.Route;
}

interface DispatchProps {
  navigate: NavigateAction;
}

interface WithProps {
  containerRef: React.RefObject<HTMLDivElement>;
}

const mapStateToProps = ({ app: { route } }: State): StateProps => ({ route });

const mapDispatchToProps = { navigate };

const containerRef = React.createRef();

export default compose<Props, {}>(
  connect<StateProps, DispatchProps, {}, State>(mapStateToProps, mapDispatchToProps),
  withProps(() => ({ containerRef })),
  lifecycle<StateProps & DispatchProps & WithProps, {}>({

    componentDidUpdate(prevProps) {
      const { current } = this.props.containerRef;

      // Scroll container to the top when route changes.
      if (this.props.route !== prevProps.route && current && current.scrollTop > 0) {
        current.scrollTo(0, 0);
      }
    },

  }),
);
