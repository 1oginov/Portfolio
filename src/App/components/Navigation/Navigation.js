/* @flow */

import * as React from 'react';

import * as R from '../../routes';
import CreatePosition from '../../views/CreatePosition';
import Home from '../../views/Home';
import Profile from '../../views/Profile';

const Navigation = ({ route }) => {
  switch (route) {
    case R.CREATE_POSITION:
      return <CreatePosition />;

    case R.PROFILE:
      return <Profile />;

    default:
      return <Home />;
  }
};

export default Navigation;
