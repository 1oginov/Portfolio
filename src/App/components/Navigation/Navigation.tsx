import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';
import {
  AccountCircleOutlined, AddCircleOutlineOutlined, BusinessCenterOutlined, PieChartOutlined,
} from '@material-ui/icons';
import * as React from 'react';

import { NavigateAction } from '../../actions';
import * as R from '../../routes';
import Analytics from '../../views/Analytics';
import ClosePosition from '../../views/ClosePosition';
import CreatePosition from '../../views/CreatePosition';
import Home from '../../views/Home';
import Position from '../../views/Position';
import Profile from '../../views/Profile';
import UpdatePosition from '../../views/UpdatePosition';

export interface Props {
  classes: { [key: string]: string };
  containerRef?: React.RefObject<HTMLDivElement>;
  navigate: NavigateAction;
  route: R.Route;
}

const Navigation: React.FunctionComponent<Props> = ({
  classes, containerRef, navigate, route,
}: Props) => {
  let Component = null;

  switch (route) {
    case R.ANALYTICS:
      Component = Analytics;
      break;

    case R.CLOSE_POSITION:
      Component = ClosePosition;
      break;

    case R.CREATE_POSITION:
      Component = CreatePosition;
      break;

    case R.POSITION:
      Component = Position;
      break;

    case R.PROFILE:
      Component = Profile;
      break;

    case R.UPDATE_POSITION:
      Component = UpdatePosition;
      break;

    default:
      Component = Home;
      break;
  }

  return (
    <>

      <div className={classes.container} ref={containerRef}>
        <Component />
      </div>

      <BottomNavigation
        className={classes.bottomNavigation}
        onChange={(event, newRoute) => navigate(newRoute)}
        showLabels
        value={route}
      >
        <BottomNavigationAction icon={<AddCircleOutlineOutlined />} label="Open" value={R.CREATE_POSITION} />
        <BottomNavigationAction icon={<BusinessCenterOutlined />} label="Portfolio" value={R.HOME} />
        <BottomNavigationAction icon={<PieChartOutlined />} label="Analytics" value={R.ANALYTICS} />
        <BottomNavigationAction icon={<AccountCircleOutlined />} label="Profile" value={R.PROFILE} />
      </BottomNavigation>

    </>
  );
};

export default Navigation;
