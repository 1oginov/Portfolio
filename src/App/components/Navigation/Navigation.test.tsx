/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import classes from 'identity-obj-proxy';
import * as React from 'react';

import Navigation from './Navigation';
import * as R from '../../routes';

it('matches snapshot if on the home page', () => {
  const wrapper = shallow(<Navigation classes={classes} navigate={() => undefined} route={R.HOME} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot if on the close position page', () => {
  const wrapper = shallow(<Navigation classes={classes} navigate={() => undefined} route={R.CLOSE_POSITION} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot if on the create position page', () => {
  const wrapper = shallow(<Navigation classes={classes} navigate={() => undefined} route={R.CREATE_POSITION} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot if on the position page', () => {
  const wrapper = shallow(<Navigation classes={classes} navigate={() => undefined} route={R.POSITION} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot if on the profile page', () => {
  const wrapper = shallow(<Navigation classes={classes} navigate={() => undefined} route={R.PROFILE} />);

  expect(wrapper).toMatchSnapshot();
});
