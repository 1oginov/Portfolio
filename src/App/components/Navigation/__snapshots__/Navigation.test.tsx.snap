// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot if on the close position page 1`] = `
<Fragment>
  <WithStyles(Connect(withHandlers(Connect(Connect(branch(FirestoreConnect(Connect(withHandlers(ClosePosition))))))))) />
  <div
    className="bottomMock"
  />
  <WithStyles(ForwardRef(BottomNavigation))
    className="bottomNavigation"
    onChange={[Function]}
    showLabels={true}
    value="CLOSE_POSITION"
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Open"
      value="CREATE_POSITION"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Portfolio"
      value="HOME"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Profile"
      value="PROFILE"
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</Fragment>
`;

exports[`matches snapshot if on the create position page 1`] = `
<Fragment>
  <WithStyles(Connect(withHandlers(withHandlers(CreatePosition)))) />
  <div
    className="bottomMock"
  />
  <WithStyles(ForwardRef(BottomNavigation))
    className="bottomNavigation"
    onChange={[Function]}
    showLabels={true}
    value="CREATE_POSITION"
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Open"
      value="CREATE_POSITION"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Portfolio"
      value="HOME"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Profile"
      value="PROFILE"
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</Fragment>
`;

exports[`matches snapshot if on the home page 1`] = `
<Fragment>
  <WithStyles(Connect(withHandlers(Connect(branch(FirestoreConnect(Connect(Connect(Home)))))))) />
  <div
    className="bottomMock"
  />
  <WithStyles(ForwardRef(BottomNavigation))
    className="bottomNavigation"
    onChange={[Function]}
    showLabels={true}
    value="HOME"
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Open"
      value="CREATE_POSITION"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Portfolio"
      value="HOME"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Profile"
      value="PROFILE"
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</Fragment>
`;

exports[`matches snapshot if on the position page 1`] = `
<Fragment>
  <WithStyles(Connect(withHandlers(Connect(Connect(branch(FirestoreConnect(Connect(Connect(withHandlers(Connect(withStateHandlers(withHandlers(withProps(Connect(lifecycle(mapProps(mapProps(Position)))))))))))))))))) />
  <div
    className="bottomMock"
  />
  <WithStyles(ForwardRef(BottomNavigation))
    className="bottomNavigation"
    onChange={[Function]}
    showLabels={true}
    value="POSITION"
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Open"
      value="CREATE_POSITION"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Portfolio"
      value="HOME"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Profile"
      value="PROFILE"
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</Fragment>
`;

exports[`matches snapshot if on the profile page 1`] = `
<Fragment>
  <WithStyles(Connect(Profile)) />
  <div
    className="bottomMock"
  />
  <WithStyles(ForwardRef(BottomNavigation))
    className="bottomNavigation"
    onChange={[Function]}
    showLabels={true}
    value="PROFILE"
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Open"
      value="CREATE_POSITION"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Portfolio"
      value="HOME"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<Memo />}
      label="Profile"
      value="PROFILE"
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</Fragment>
`;
